rotkehlchen/chain/substrate/manager.py

```diff
@@ -0,0 +1,555 @@
+import logging
+from functools import wraps
+from http import HTTPStatus
+from json.decoder import JSONDecodeError
+from typing import Any, Callable, Dict, Iterable, List, Optional, Tuple, cast
+
+import requests
+from requests.adapters import Response
+from substrateinterface import SubstrateInterface
+from substrateinterface.exceptions import SubstrateRequestException
+from typing_extensions import Literal
+
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.constants.misc import ZERO
+from rotkehlchen.errors import RemoteError, UnknownAsset
+from rotkehlchen.fval import FVal
+from rotkehlchen.greenlets import GreenletManager
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.typing import SubstrateAddress
+from rotkehlchen.user_messages import MessagesAggregator
+from rotkehlchen.utils.serialization import rlk_jsonloads_dict
+
+from .typing import (
+    BlockNumber,
+    Chain,
+    ChainId,
+    ChainProperties,
+    DictNodeNameNodeAttributes,
+    NodeInterface,
+    NodeName,
+    NodeNameAttributes,
+    NodesCallOrder,
+)
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+"""
+Polkadot/Kusama integration
+------------------
+General issue:
+https://github.com/rotki/rotki/issues/1081
+
+Milestone 1 issues:
+https://github.com/rotki/rotki/issues/1919
+https://github.com/rotki/rotki/issues/1920
+
+Milestone 2 issues:
+
+Related issues:
+https://github.com/rotki/rotki/issues/2032 (ENS support)
+
+TODO priority 1:
+! Make sure tests do not connect to nodes per test
+- Implement tests for:
+  * API blockchain (CRUD KSM account)
+  * API blockchain query KSM account balance
+  * SubstrateManager (check EthereumManager ones as well)
+
+TODO priority 2:
+- Issue 1920: add ability to use a local Kusama node
+    * https://github.com/rotki/rotki/issues/1920
+    * Either include it in KusamaNodeName or consider to use a generic structure
+    for any substrate local node.
+- Issue Add support for ENS addresses
+
+TODO priority 3:
+- Find/ask for a contract that allows batch queries for balances
+"""
+
+
+def request_available_nodes(func: Callable[..., Any]) -> Callable[..., Any]:
```

Comment:
It would be nice if the node order here could be customized per call. Does not need to happen in this PR, but this is something that we needed in the past many times with ethereum

Reply:
I've always had in mind to allow the customisation once I tackle the "own node" issue.